src/datatypes/src/vectors/constant.rs

```diff
@@ -83,6 +84,34 @@ impl ConstantVector {
             self.length,
         )))
     }
+
+    pub(crate) fn take_vector(&self, indices: &UInt32Vector) -> Result<VectorRef> {
+        if indices.is_empty() {
+            return Ok(self.slice(0, 0));
+        }
+        let len = self.len() as u32;
+        indices
+            .iter_data()
+            .map(|idx| {
+                if let Some(idx) = idx {
+                    ensure!(
+                        idx < len,
+                        error::BadArrayAccessSnafu {
+                            index: idx as usize,
```

Comment:
suggestion: We could cast idx to usize instead of casting len to u32
```suggestion
        let len = self.len();
        indices
            .iter_data()
            .map(|idx| {
                if let Some(idx) = idx {
                    ensure!(
                        idx as usize < len,
                        error::BadArrayAccessSnafu {
                            index: idx as usize,
```